core/src/main/java/org/infinispan/factories/ComponentRegistry.java

```diff
@@ -217,6 +221,29 @@ public void start() {
       super.start();
    }
 
+   @Override
+   protected CompletionStage<Void> await() {
+      if (configuration.simpleCache() || icr.running().isInternalCache(cacheName)) return super.await();
```

Comment:
```suggestion
      if (configuration.simpleCache() || icr.running().isInternalCache(cacheName)) return super.await();
```
You can use this method from CacheMode instead. https://github.com/infinispan/infinispan/blob/main/core/src/main/java/org/infinispan/configuration/cache/CacheMode.java#L109